### DISCUSSION TITLE
Latex `\tag`-like functionality for comments in equations

### DISCUSSION THREAD

**Discussion by mtoohey31** [typst/typst#3901] [Q&A]
While this might not be what it was intended for, when using Latex I've often used `\tag` within long equations to comment on things, such as why an equality is justified. For example, I'd use:

```latex
\begin{align*}
  J^{\alpha\beta}_{\text{reg}}
  &=\frac1{2N}\sum_{i=1}^N\left(y^{(i)}-t^{(i)}\right)^2+\sum_{j=1}^D\alpha_j|w_j|+\frac12\sum_{j=1}^D\beta_jw_j^2\tag{by definition}\\
  &=\frac1{2N}\sum_{i=1}^N\left(\sum_{j=1}^Dw_jx^{(i)}_j+b-t^{(i)}\right)^2+\sum_{j=1}^D\alpha_j|w_j|+\frac12\sum_{j=1}^D\beta_jw_j^2\tag{by definition of $y^{(i)}$ in our model}
\end{align*}
```

...to create:

![tag](https://github.com/typst/typst/assets/36740602/6b66c9e2-636c-45a1-812b-4f11b876b5e1)

I've tried using this so far:

```typst
#let tag = it => $& #par[(#it)]$
```

...which isn't too bad, but it doesn't right-align the text, and doesn't have the wrapping behaviour seen in the image above where the text moves underneath when the equation and text together on one line is too wide.

Does anyone have suggestions about how I could do better? Or maybe there's a package that already does this?

**Comment by qwinsi** [comment]
This topic is still relevant...
Any suggestions?

**Comment by qwinsi** [comment]
Now I'm using the following code snippet for a walkaround
```typst
#let named_equations(content, name) = {
  let numbering_func(it) = [(#name)];
  math.equation(block: true, numbering: numbering_func, content)
}

#named_equations($
nabla times bold(H) &= bold(J) +(diff bold(D))/(diff t) \
nabla times bold(E) &= -(diff bold(B))/(diff t) \
nabla dot.op bold(B) &= 0 \
nabla dot.op bold(D) &= rho
$, [Maxwell's equations])
``` 

![image](https://github.com/user-attachments/assets/7b627e06-6f82-473d-b64f-3bb7614d9164)

Please let me know if there are any more proper ways to do so...

**Reply by michaelfortunato** [reply]
Curious about this as well

**Reply by Soham-Saha** [reply]
Nice! I slightly modified your function:

```typst
#let tag(name, body) = {
  math.equation(block: true, numbering: x => name, body)
}

// Use as #tag[(Beautiful)][$e^(i pi)+1=0$]
```

**Comment by cprieto** [comment]
So I guess there is not an easy way to do this yet?

**Comment by kalsifer-742** [comment]
Personally I use the [Alignment](https://typst.app/docs/reference/math/).
I use the right aligned column for comments.
Like this:
<img width="301" height="157" alt="image" src="https://github.com/user-attachments/assets/5900888a-c3da-40af-b633-76439422290e" />
